OpenRA.Mods.Common/Activities/Move/AttackMoveActivity.cs

```diff
@@ -18,31 +19,70 @@ namespace OpenRA.Mods.Common.Activities
 {
 	public class AttackMoveActivity : Activity
 	{
-		const int ScanInterval = 7;
-
+		readonly Func<Activity> getInner;
 		Activity inner;
-		int scanTicks;
+		Activity attack;
 		AutoTarget autoTarget;
+		bool moving;
 
-		public AttackMoveActivity(Actor self, Activity inner)
+		public AttackMoveActivity(Actor self, Func<Activity> getInner)
 		{
-			this.inner = inner;
+			this.getInner = getInner;
```

Comment:
Style nit / question: is `inner`/`getInner` always going to be a move activity in practice? If yes, then IMO it would be helpful for readability if these were renamed to `move` / `getMove`

Reply:
In practice it is either Move or Follow.